azure/services/async/async.go

```diff
@@ -230,3 +244,19 @@ func getRetryAfterFromError(err error) time.Duration {
 	}
 	return ret
 }
+
+// ContextDeadlineExceededOrCanceled checks if it's an intermittent context deadline
+// exceeded or canceled error.
+func ContextDeadlineExceededOrCanceled(ctx context.Context, err error) bool {
```

Comment:
Just thinking if it is appropriate to keep this function here or move to azure/errors.go or somewhere else.
But then, I think it is better to keep it here until we find the need to use it at other place and at that time we will have better clarity to find the right home for this.